crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -35,15 +47,56 @@ use std::{
     sync::atomic::{AtomicI64, Ordering},
 };
 
-/// Lightweight unique ID of an entity.
+/// Lightweight identifier of an [entity](crate::entity).
+///
+/// The identifier is implemented using a generational index: a combination of an ID and a
+/// generation. This allows fast insertion after data removal in an array while minimizing loss of
+/// spatial locality.
+///
+/// # Usage
+///
+/// This data type is returned by iterating a `Query` ([learn more]). It can also be obtained by calling [`EntityCommands::id`] or [`EntityMut::id`].
```

Comment:
```suggestion
/// This data type is returned by iterating a `Query` ([learn more]) with a `Entity` [`WorldQuery`] argument. It can also be obtained by calling [`EntityCommands::id`] or [`EntityMut::id`].
```